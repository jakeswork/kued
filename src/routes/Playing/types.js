import { shape } from 'prop-types';

export const propTypes = {
  classes: shape({}).isRequired,
};

export const defaultProps = {

};
