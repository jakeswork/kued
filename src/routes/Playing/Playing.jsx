import React from 'react';
import { Tabs, TabList } from 'react-tabs';

import { propTypes, defaultProps } from './types';
import MainLayout from '../../layouts/MainLayout';
import Text from '../../components/Text';
import TabHeading from './components/TabHeading';
import WarmaneTab from './components/WarmaneTab';

const Playing = ({ classes }) => (
  <MainLayout>
    <Text h1>Playing</Text>
    <Tabs selectedTabClassName={classes.tabActive}>
      <TabList className={classes.tabList}>
        <TabHeading>Warmane</TabHeading>
      </TabList>
      <WarmaneTab />
    </Tabs>
  </MainLayout>
);

Playing.propTypes = propTypes;

Playing.defaultProps = defaultProps;

export default Playing;
