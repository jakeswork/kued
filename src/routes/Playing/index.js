import withStyles from 'react-jss';

import Playing from './Playing';
import styles from './styles';

export default withStyles(styles)(Playing);
