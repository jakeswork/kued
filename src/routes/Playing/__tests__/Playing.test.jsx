import React from 'react';
import { shallow } from 'enzyme';

import Playing from '../Playing';

const defaultProps = {
  classes: {},
};

describe('The Playing instance', () => {
  describe('When called with default props', () => {
    it('should render without throwing an error', () => {
      const wrapper = shallow(<Playing {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
