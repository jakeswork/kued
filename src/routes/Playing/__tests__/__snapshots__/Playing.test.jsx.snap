// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Playing instance When called with default props should render without throwing an error 1`] = `
<Jss(MainLayout)>
  <Jss(Text)
    bold={false}
    caption={false}
    h1={true}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    link={null}
    style={Object {}}
  >
    Playing
  </Jss(Text)>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <TabList
      className="react-tabs__tab-list"
    >
      <Jss(TabHeading)
        sub={false}
      >
        Warmane
      </Jss(TabHeading)>
    </TabList>
    <Jss(WarmaneTab) />
  </Tabs>
</Jss(MainLayout)>
`;
